pkg/core/types.go

```diff
@@ -51,3 +53,32 @@ type configObject struct {
 	allKeys  bool
 	keys     map[string]struct{}
 }
+
+type podController interface {
+	runtime.Object
+	metav1.Object
+	GetObject() runtime.Object
+	GetPodTemplate() *corev1.PodTemplateSpec
+	SetPodTemplate(*corev1.PodTemplateSpec)
+	DeepCopy() podController
+}
+
+type Deployment struct {
```

Comment:
I think this `Deployment` type could be private? If it's implementing a private interface I'm not sure it should be public, best practices and all that 😅